src/pages/en/guides/markdown-content.md

```diff
@@ -419,6 +395,97 @@ By default, Astro comes with [GitHub-flavored Markdown](https://github.com/remar
     };
     ```
 
+### Injecting frontmatter
+
+You may want to add frontmatter properties to your markdown files programmatically. By using a [remark or rehype plugin](#markdown-plugins), you can generate these properties based on a file's contents.
+
+You can append to the `data.astro.frontmatter` property from your plugin's `file` argument like so:
+
+```js
+// example-remark-plugin.js
+export function exampleRemarkPlugin() {
+  // All remark and rehype plugins return a separate function
+  return function (tree, file) {
+    file.data.astro.frontmatter.customProperty = 'Generated property';
+  }
+}
+```
+
+After applying this plugin to your `markdown` config:
+
+```js
+// astro.config.mjs
+import { exampleRemarkPlugin } from './example-remark-plugin';
+
+export default {
+  markdown: {
+    remarkPlugins: [exampleRemarkPlugin],
+  },
+};
+
+```
+
+...every markdown file will have `customProperty` in its frontmatter! This is available when [importing your markdown](#importing-markdown) and from [the `Astro.props.frontmatter` property in your layouts](#markdown-layouts).
+
+#### Example: calculate reading time
+
+You can use a [remark plugin](https://github.com/remarkjs/remark) to add a reading time to your frontmatter. We recommend a couple helper packages:
```

Comment:
```suggestion
You can use a [remark plugin](https://github.com/remarkjs/remark) to add a reading time to your frontmatter. We recommend two helper packages:
```
(Simplified language is generally easier for a wider audience while following instructions.)